### ISSUE TITLE
Use the inputs provided by WooCommerce checkout block

### ISSUE THREAD

**Issue by LevinMedia** [Automattic/woocommerce-payments#1152]
When rendering the checkout fields in the new WooCommerce Checkout block, Update the credit card forms to use the fields provided by the block. 

**Stripe example example:**
<img width="722" alt="Screen Shot 2020-12-07 at 1 14 08 PM" src="https://user-images.githubusercontent.com/4500952/101406444-53f4dd80-388e-11eb-876a-b909a3b471e4.png">

**WooCommerce payments for comparison:**
<img width="724" alt="Screen Shot 2020-12-07 at 1 14 20 PM" src="https://user-images.githubusercontent.com/4500952/101406544-725ad900-388e-11eb-9c96-66d1103f59a3.png">

It's particularly challenging when the theme has a dark background

**Comment by LevinMedia** [comment]
@RadoslavGeorgiev @vbelolapotkov I've added priority high, as this can affect usability pretty significantly.

**Comment by vbelolapotkov** [comment]
@RadoslavGeorgiev that looks like a simple fix, unless I miss something unobvious complexity. Maybe worth assigning a good first issue label and ask the new team member to work on it?

**Comment by RadoslavGeorgiev** [comment]
@LevinMedia I took a look at both gateways and indeed, Stripe works better with darker themes, while WCPay can actually be unusable (example when using the `spice-software-dark` theme, the first dark theme I stumbled upon):

<img width="647" alt="image" src="https://user-images.githubusercontent.com/5311119/152313367-7219dc82-15f2-4796-a374-587371d1b8f5.png">

The CC number field is indistinguishable from the date fields, because this example uses UPE, a single element 😨

I was curious and took a look at the code we have in the Stripe gateway, which is basically just a [few CSS rules here](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/develop/assets/js/stripe.js#L74-L92). There are a couple of filters (namely `wc_stripe_elements_styling` and `wc_stripe_elements_classes`) in the gateway, which allow theme developers to customize the appearance of the fields, but most themes don't utilize them.

The CSS from Stripe's example does not even include a background color, which surprised me. After further examination, it turns out that the checkout block has Stripe-specific styles:

```css
.wc-block-gateway-container .wc-block-gateway-input {
    font-size: 1em;
    line-height: 1.375;
    background-color: #fff;
    padding: .75em 0 .75em 16px;
    border-radius: 4px;
    border: 1px solid #50575e;
    width: 100%;
    font-family: inherit;
    margin: 0;
    box-sizing: border-box;
    height: 3em;
    color: #2b2d2f;
    cursor: text;
}
```

So basically fields have a white background, which disappears when using a light theme, but makes the fields usable with dark themes. Should we apply the same approach to WCPay? If the answer is yes, then this is a simple fix @vbelolapotkov .

**Comment by vbelolapotkov** [comment]
I think yes, we should. I'd only double-check the color values are taken from the standard woo/wp colors.

**Comment by LevinMedia** [comment]
@vbelolapotkov @RadoslavGeorgiev if work hasn't begun on this already, I'm on board with that approach.
